frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useState } from "react";
+
+import {
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  Button,
+  useToast,
+} from "@chakra-ui/react";
+import {
+  Camper,
+  EditCamperInfoFields,
+  EditModalSetterFunctions,
+} from "../../../../types/CamperTypes";
+import { FormQuestion } from "../../../../types/CampsTypes";
+import CamperApiClient from "../../../../APIClients/CamperAPIClient";
+import EditCamperModalForm from "./EditCamperModalForm";
+
+type EditCamperModalProps = {
+  camper: Camper;
+  editCamperModalIsOpen: boolean;
+  formQuestions: FormQuestion[];
+  editCamperOnClose: () => void;
+};
+
+const EditCamperModal = ({
+  camper,
+  formQuestions,
+  editCamperModalIsOpen,
+  editCamperOnClose,
+}: EditCamperModalProps): JSX.Element => {
+  const toast = useToast();
+
+  // State for the edit form fields
+  const [firstName, setFirstName] = useState<string>(camper.firstName);
+  const [lastName, setLastName] = useState<string>(camper.lastName);
+  const [age, setAge] = useState<number>(camper.age);
+  const [allergies, setAllergies] = useState<string | undefined>(
+    camper.allergies,
+  );
+  const [specialNeeds, setSpecialNeeds] = useState<string | undefined>(
+    camper.specialNeeds,
+  );
+  const [formResponses, setFormResponses] = useState<
+    Map<string, string> | undefined
+  >(camper.formResponses);
+
+  const setStateFuncs: EditModalSetterFunctions = {
+    setFirstName,
+    setLastName,
+    setAge,
+    setAllergies,
+    setSpecialNeeds,
+    setFormResponses,
+  };
+
+  const stateVariables: EditCamperInfoFields = {
+    hasPaid: camper.hasPaid,
+    firstName,
+    lastName,
+    age,
+    allergies,
+    specialNeeds,
+    formResponses,
+  };
+
+  const onSaveClicked = async () => {
+    const editCamperFields: EditCamperInfoFields = {
+      firstName,
+      lastName,
+      age,
+      allergies,
+      specialNeeds,
+      formResponses,
+      hasPaid: camper.hasPaid,
+    };
+    const isEditCamperSuccess = await CamperApiClient.updateCampersById(
+      [camper.id],
+      editCamperFields,
+    );
+
+    if (isEditCamperSuccess) {
+      toast({
+        description: `${camper.firstName} ${camper.lastName}'s information has been updated`,
+        status: "success",
+        variant: "subtle",
+        duration: 3000,
+      });
+
+      setTimeout(() => {
+        window.location.reload();
```

Comment:
I know this is temporary, but if you want a non-reload solution you can check out what I did https://github.com/uwblueprint/focus-on-nature/pull/129
basically I'm triggering a refetch by changing some state in the useEffect's dependency array and I'm sure you could do that here too, it's still not ideal but it's better than an entire page reload (even temporarily).

Reply:
That's completely fair but to refresh the state, I would need to change some dependancy in the CampOverview's useEffect so that it can get all the camp data again. I think that could be a little messy and since we're gonna replace with global state anyways, I think it's fine as is for now just to get the ticket moving. 

Good call out though- hadn't thought of that for this use case

Reply:
Sounds good.

Reply:
How about something like an overall `sessionDidChange`? I suggest it in https://github.com/uwblueprint/focus-on-nature/pull/128/files# anyways

https://github.com/uwblueprint/focus-on-nature/blob/29c5d58a7f3726d57a5d823119ae6f488c566602/frontend/src/components/pages/CampOverview/index.tsx#L65-L74